src/Altinn.App.Core/Expressions/ExpressionEvaluator.cs

```diff
@@ -0,0 +1,307 @@
+using System.Globalization;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+
+using Altinn.App.Core.Models.Expressions;
+using Altinn.App.Core.Models.Layout.Components;
+
+namespace Altinn.App.Core.Expressions;
+
+/// <summary>
+/// Static class used to evaluate expressions. Holds the implementation for all expression functions.
+/// </summary>
+public static class ExpressionEvaluator
+{
+    /// <summary>
+    /// Shortcut for evaluating a boolean expression on a given property on a <see cref="BaseComponent" />
+    /// </summary>
+    public static bool EvaluateBooleanExpression(LayoutEvaluatorState state, ComponentContext context, string property, bool defaultReturn)
+    {
+        var expr = property switch
+        {
+            "hidden" => context.Component.Hidden,
+            "required" => context.Component.Required,
+            _ => throw new Exception($"unknown boolean expression property {property}")
+        };
+        if (expr is null)
+        {
+            return defaultReturn;
+        }
+
+        return EvaluateExpression(state, expr, context) switch
+        {
+            true => true,
+            false => false,
+            null => defaultReturn,
+            _ => throw new Exception($"Return from evaluating \"{property}\" on \"{context.Component.Id}\" was not boolean (value)")
```

Comment:
We should either find a more specific exception or create a custom application exception.

Reply:
I created `ExpressionEvaluatorTypeErrorException` and `DataModelException` to comply with the coding standard.